docs/RELEASING.md

```diff
@@ -1,42 +1,43 @@
 # Releasing the Connectors project
 
-The Connectors project is packaged as a Gem that is added in the Enterprise Search distribution.
+The Connectors project contains logic that is also packaged as a Gem to be included in the Enterprise Search distribution ([example](https://github.com/elastic/ent-search/blob/9e87bf1c293df97f3a550ed1e2a7efac18c9d8f2/Gemfile#L262)).
 
-The version scheme we use is **MAJOR.MINOR.PATCH.BUILD** and stored in the [VERSION](VERSION) file
+The version scheme we use is **MAJOR.MINOR.PATCH.BUILD** and stored in the [VERSION](https://github.com/elastic/connectors/blob/main/VERSION) file
 at the root of this repository.
 
 ## RubyGem Account
 
 When releasing Gems, you will be asked for an Email and Password.
 Look into the Vault in the `ent-search-team/rubygem` secret.
 
-
 ## Unified release
 
 **MAJOR.MINOR.PATCH** should match the Elastic and Enterprise Search version it targets 
-and the *BUILD* number should be set to **0** the day the release is created
+and the *BUILD* number should be set to **0** the day the Connectors release is created
 to be included with the Enterprise Search distribution.
 
 For example, when shipping for `8.1.2`, the version is `8.1.2.0`.
 
 To release Connectors:
 
-- set the VERSION file to the right version
-- make sure all tests and linter pass with `make lint test`
-- run `make release`
+- Set the VERSION file to the new/incremented version on the release branch
+- Make sure all tests and linter pass with `make lint test`
+- PR these changes to the appropriate Connectors release branch
+- Run `make release`
 
-A Gem file will be created in the `.gem` directory, that needs to be embed in Enterprise Search.
+A Gem will be published to RubyGems: https://rubygems.org/gems/connectors_sdk.
 
-Immediately update ent-search's gem dependency with the released version:
+Then immediately update ent-search's gem dependency with the released version:
 
-- update the Gem in vendor/cache
-- update Gemfile
+- Update Gemfile if necessary
+- Run `./script/bundle update connectors_sdk`
+- Verify bundler picked up the version you expected it to update to
+- PR these changes to the appropriate Enterprise Search release branch ([example](https://github.com/elastic/ent-search/pull/6476))
 
 Last, take care of the branching:
```

Comment:
Reviewing this made me realize that we need another step, which is updating the VERSION in the branch that just got released. PR for this time: https://github.com/elastic/connectors/pull/81